pod/perlre.pod

```diff
@@ -1593,35 +1674,41 @@ following this construct will be numbered as though the construct
 contained only one branch, that being the one with the most capture
 groups in it.
 
-This construct is useful when you want to capture one of a
-number of alternative matches.
+This construct is useful when you want to capture one of a number of
+alternative matches.
 
-Consider the following pattern.  The numbers underneath show in
-which group the captured content will be stored.
+Consider the following pattern. The numbers underneath show in which
+group the captured content will be stored.
 
 
     # before  ---------------branch-reset----------- after
     / ( a )  (?| x ( y ) z | (p (q) r) | (t) u (v) ) ( z ) /x
     # 1            2         2  3        2     3     4
 
-Be careful when using the branch reset pattern in combination with
-named captures. Named captures are implemented as being aliases to
-numbered groups holding the captures, and that interferes with the
-implementation of the branch reset pattern. If you are using named
-captures in a branch reset pattern, it's best to use the same names,
-in the same order, in each of the alternations:
+Prior to Perl 5.37.8 it was necessary to be careful when using the
+branch reset pattern in combination with named captures and named
+recursion. Named captures were implemented as being aliases to the
+underlying numbered capture group, which meant that if the same name
+were used in different alternations inside of a branch reset construct
+that the named buffer might end up referecing the numbered buffer in a
+confusing way. In older perls if you are using named captures in a
+branch reset pattern, it's best to use the same names, in the same
+order, in each of the alternations:
 
    /(?|  (?<a> x ) (?<b> y )
       |  (?<a> z ) (?<b> w )) /x
 
-Not doing so may lead to surprises:
+not doing so could lead to surprises:
 
   "12" =~ /(?| (?<a> \d+ ) | (?<b> \D+))/x;
   say $+{a};    # Prints '12'
   say $+{b};    # *Also* prints '12'.
 
 The problem here is that both the group named C<< a >> and the group
-named C<< b >> are aliases for the group belonging to C<< $1 >>.
+named C<< b >> are aliases for the group belonging to C<< $1 >>. As of
```

Comment:
This doesn't make sense to me.  I think you mean

> named C<< b >>, prior to Perl 5.37.8, were aliases for the group belonging to C<< $1 >>. As of

Reply:
Yep, that is better wording.